process/transactions/datafield/parser_test.go

```diff
@@ -0,0 +1,199 @@
+package datafield
+
+import (
+	"testing"
+
+	indexer "github.com/ElrondNetwork/elastic-indexer-go"
+	"github.com/ElrondNetwork/elastic-indexer-go/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func createMockArgumentsOperationParser() *ArgsOperationDataFieldParser {
+	return &ArgsOperationDataFieldParser{
+		PubKeyConverter:  &mock.PubkeyConverterMock{},
+		Marshalizer:      &mock.MarshalizerMock{},
+		ShardCoordinator: &mock.ShardCoordinatorMock{},
+	}
+}
+
+func TestNewOperationDataFieldParser(t *testing.T) {
+	t.Parallel()
+
+	t.Run("NilMarshalizer", func(t *testing.T) {
```

Comment:
might add `t.Parallel()` to all inner tests. otherwise, only the `TestNewOperationDataFieldParser` will be run in parallel and all the inner tests will run sequentially

Reply:
added